tools/install-monorepo.sh

```diff
@@ -15,143 +17,344 @@ function usage {
+		# initially, we need to source it.
+		if [[ -f "$(brew --prefix nvm)/nvm.sh" ]]; then
+			# Sometimes nvm is installed via brew, which is an unsupported install
+			# route, and uses a wrapper helper script.
+			. "$(brew --prefix nvm)/nvm.sh"
+		else
+			export NVM_DIR="$HOME/.nvm" && . "$NVM_DIR"/nvm.sh --no-use
+		fi
+
+		if [[ v$(nvm --version) != "$good_nvm_version" ]]; then
+			echo "nvm: wrong version"
+			echo "Updating nvm..."
+
+			# Download latest version
+			curl -s -o- "https://raw.githubusercontent.com/nvm-sh/nvm/$good_nvm_version/install.sh" | bash - &>/dev/null &&
+
+			# Source again to get latest shell functions
+			. "$NVM_DIR/nvm.sh" --no-use
+
+			# Something's wrong, so abort.
+			if [[ "v$(nvm --version)" != "$good_nvm_version" ]]; then
+				die "Unable to update nvm!"
 			fi
+			echo "Updated nvm."
+		fi
+	else
+		echo "nvm: not found"
+		info "Installing nvm..."
+		curl -s -o- https://raw.githubusercontent.com/nvm-sh/nvm/$good_nvm_version/install.sh | bash - &>/dev/null &&
+
+		# Source to get latest shell functions
+		export NVM_DIR="$HOME"/.nvm && . "$NVM_DIR"/nvm.sh --no-use
+
+		# Something's wrong, so abort.
+		if ! has_command nvm; then
+			die "Unable to install nvm!"
 		fi
+		echo "Installed nvm."
 	fi
-fi
 
-source .github/versions.sh
-info "Installing and linking PHP $PHP_VERSION"
-brew install php@$PHP_VERSION
-brew link php@$PHP_VERSION
+	echo "nvm: available"
+}
+
+# Checks for and installs node.
+function do_node {
+	good_node_version=$(cat .nvmrc)
+	if ! has_command node; then
+		echo "node: not found"
+		echo "Installing node..."
+		nvm install
+		# Something's wrong, so abort.
+		if ! has_command node; then
+			die "Unable to install node!"
+		fi
+		echo "Node installed."
+	elif [[ $(node --version) != v"$good_node_version" ]]; then
+		# Install correct version if not yet installed
+		if ! nvm list node | grep -q "$good_node_version"; then
+			echo "node: needed version not installed"
+			echo "Installing correct node version..."
+			nvm install
+			# Something's wrong, so abort.
+			if ! nvm list node | grep -q "$good_node_version"; then
+				die "Unable to needed version of node!"
+			fi
+			echo "Correct version of node installed."
+		fi
+	fi
+	# Switch node version
+	nvm use &> /dev/null
+	echo "node: available"
+}
 
-info "Checking composer..."
-if ! command -v composer &>/dev/null; then
-	info "Installing Composer"
-	EXPECTED_CHECKSUM="$(php -r 'copy("https://composer.github.io/installer.sig", "php://stdout");')"
-	php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
-	ACTUAL_CHECKSUM="$(php -r "echo hash_file('sha384', 'composer-setup.php');")"
+function do_bash {
+	if [[ -z "${BASH_VERSINFO}" || -z "${BASH_VERSINFO[0]}" || ${BASH_VERSINFO[0]} -lt 4 ]]; then
+		echo "bash 4+: not found"
+		echo "Installing bash..."
+		brew install bash
+		if [[ -z "${BASH_VERSINFO}" || -z "${BASH_VERSINFO[0]}" || ${BASH_VERSINFO[0]} -lt 4 ]]; then
```

Comment:
This isn't going to work, it's still testing the currently-running version of bash rather than the one just installed.

Something like this should work.
```suggestion
		if bash -c '[[ -z "${BASH_VERSINFO}" || -z "${BASH_VERSINFO[0]}" || ${BASH_VERSINFO[0]} -lt 4 ]]'; then
```